File: ProResults/ProResults/Models/Badge.cs
using System.ComponentModel.DataAnnotations;

namespace ProResults.Models
{
    public class Badge
    {
        public Guid Id { get; set; }
        
        [Required]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        public string Issuer { get; set; } = string.Empty;
        
        public DateTime IssuedDate { get; set; }
        
        public DateTime? ExpirationDate { get; set; }
        
        public string? ImageUrl { get; set; }
        
        [Required]
        public string CredentialJson { get; set; } = string.Empty;
        
        public bool IsVerified { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Foreign key
        public Guid UserId { get; set; }
        public User User { get; set; } = null!;
    }
}

File: frontend/src/types/badge.ts
export interface Badge {
  id: string;
  name: string;
  description: string;
  issuer: string;
  issuedDate: string;
  expirationDate?: string;
  imageUrl?: string;
  credentialJson: string;
  isVerified: boolean;
}

export interface OpenBadgeCredential {
  '@context': string[];
  type: string[];
  id: string;
  issuer: {
    id: string;
    type: string;
    name: string;
  };
  validFrom: string;
  validUntil?: string;
  credentialSubject: {
    id: string;
    type: string[];
    achievement: {
      id: string;
      type: string[];
      name: string;
      description: string;
      criteria: {
        narrative: string;
      };
      image?: {
        id: string;
        type: string;
      };
    };
  };
  proof?: {
    type: string;
    created: string;
    verificationMethod: string;
    proofPurpose: string;
    proofValue: string;
  };
}



export interface LoginResponse {
  token: string;
  user: {
    id: string;
    email: string;
    name: string;
  };
}
